cmd/tui/selectable_table.go

```diff
@@ -0,0 +1,475 @@
+	for i, row := range s.rawRows {
+		if i == int(remove) {
+			delete(s.activeRows, i)
+			delete(s.disabledRows, i)
+
+			continue
+		}
+
+		col := make([]string, 0, len(row))
+		col = append(col, row...)
+		newRows = append(newRows, col)
+	}
+
+	s.rawRows = newRows
+	s.filterRows(true)
+
+	return s, nil
+}
+
+func (s *selectableTable) handleInsertEvent(insert InsertMsg) (tea.Model, tea.Cmd) {
+	s.rawRows = append(s.rawRows, insert)
+	s.filterRows(false)
+
+	return s, nil
+}
+
+func (s *selectableTable) handleDisableEvent(disable DisableMsg) (tea.Model, tea.Cmd) {
+	// indexes to disable should always be absolute indexes and do not need to go through s.filterMap.
+	delete(s.activeRows, int(disable))
+	s.disabledRows[int(disable)] = true
+
+	return s, nil
+}
+
+func (s *selectableTable) handleEnableEvent(enable EnableMsg) (tea.Model, tea.Cmd) {
+	// indexes to enable should always be absolute indexes and do not need to go through s.filterMap.
+	delete(s.disabledRows, int(enable))
+
+	return s, nil
+}
+
+func (s *selectableTable) handleKeyEvent(key tea.KeyMsg) (tea.Model, tea.Cmd) {
+	switch key.Type {
+	case tea.KeyEsc:
+		fallthrough
+	case tea.KeyCtrlC:
+		s.activeRows = map[int]bool{}
+		fallthrough
+	case tea.KeyEnter:
+		s.active = false
+		return s, tea.Quit
+
+	case tea.KeyUp:
+		if s.currentRow > 0 {
+			s.currentRow--
+
+			if s.currentRow < s.startIndex {
+				s.startIndex--
+			}
+		}
+	case tea.KeyDown:
+		if s.currentRow < len(s.formatRows)-1 {
+			s.currentRow++
+
+			if s.currentRow > s.startIndex+(s.size-1) {
+				s.startIndex++
+			}
+		}
+	case tea.KeyLeft:
+		s.activeRows = map[int]bool{}
+	case tea.KeyRight:
+		for i := range s.formatRows {
+			if !s.disabledRows[s.filterMap[i]] {
+				s.activeRows[s.filterMap[i]] = true
+			}
+		}
+
+	case tea.KeySpace:
+		if !s.disabledRows[s.filterMap[s.currentRow]] && len(s.formatRows) > 0 {
+			if s.activeRows[s.filterMap[s.currentRow]] {
+				delete(s.activeRows, s.filterMap[s.currentRow])
+			} else {
+				s.activeRows[s.filterMap[s.currentRow]] = true
+			}
+		}
+
+	case tea.KeyBackspace:
+		if len(s.filter) > 0 {
+			s.filter = s.filter[:len(s.filter)-1]
+		}
+
+		s.filterRows(true)
+
+	case tea.KeyDelete:
+		if len(s.filter) > 0 {
+			s.filter = s.filter[1:len(s.filter)]
+		}
+
+		s.filterRows(true)
+	case tea.KeyRunes:
+		for _, rune := range key.Runes {
+			s.filter += string(rune)
+		}
+
+		s.filterRows(true)
+	default:
+		s.activeRows = map[int]bool{}
+		s.active = false
+		if s.err != nil {
+			s.err = fmt.Errorf("Input cancelled")
+		}
+
+		return s, tea.Quit
+	}
+
+	return s, nil
+}
+
+func (s *selectableTable) rowStyle(row int) {
+	for col := range s.formatRows[row] {
+		rawRowIndex := s.filterMap[row]
+		textStyle := lipgloss.NewStyle().SetString(s.rawRows[rawRowIndex][col])
+		if row == s.currentRow {
+			if s.activeRows[s.filterMap[row]] {
+				textStyle = textStyle.Bold(true).Foreground(Green)
+			} else if s.disabledRows[s.filterMap[row]] {
```

Comment:
So we don't use it in MicroCloud yet, I thought this could be something we can extend later to block joiners whose websockets have closed. 

You can simulate it yourself by populating a `SelectableTable` and then sending `table.SendUpdate(DisableMsg(i))` where `i` is the row index to disable.

Reply:
Ah ok that explains why I haven't found a related section in the code that makes use of it. 
As it's currently unused code, should we comment it out or rather add a note for the future?

Reply:
I think a TODO is in order here. I'd rather not comment out chunks of code from different places.